From fedfb81634b3e7880d8e6d4445314ec89bd3ee42 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 18 Sep 2018 23:15:44 +0530
Subject: [PATCH 0410/1852] dt-bindings: display: xlnx: Add support for
 ancillary port

commit 3600c1809262b72789c025babc66c467981d366d from
https://github.com/Xilinx/linux-xlnx.git

Add support for ancillary port to be connected. This port is currently
used for adding audio to SDI Tx. Based on this extra port, the SDI Tx IP
may set the "Use Ancillary Data" bit.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/display/xlnx/xlnx,sdi-tx.txt     | 21 +++++++++++++++----
 1 file changed, 17 insertions(+), 4 deletions(-)

diff --git a/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt b/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt
index dcdb3597d321..54701ed3098f 100644
--- a/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt
+++ b/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt
@@ -11,6 +11,8 @@ Required properties:
    external device. In the display controller port nodes, topology
    for entire pipeline should be described using the DT bindings defined in
    Documentation/devicetree/bindings/graph.txt.
+   Minimum one port is required. At max, 2 ports are present.
+   The reg index for AXI4 stream port is 0 and for ancillary data is 1.
 
 Optional properties:
  - xlnx,vpss: vpss phandle
@@ -26,10 +28,21 @@ Example:
 		#address-cells = <1>;
 		#size-cells = <0>;
 		xlnx,vpss = <&v_proc_ss_0>;
-		encoder_sdi_port: port@0 {
-			reg = <0>;
-			sdi_encoder: endpoint {
-				remote-endpoint = <&xyz_port>;
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			encoder_sdi_port: port@0 {
+				reg = <0>;
+				sdi_encoder: endpoint {
+					remote-endpoint = <&pl_disp_crtc>;
+				};
+			};
+
+			sdi_audio_port: port@1 {
+				reg = <1>;
+				sdi_audio_sink_port: endpoint {
+					remote-endpoint = <&sditx_audio_embed_src_port>;
+				};
 			};
 		};
 	};
-- 
2.31.1

